pkg/js/util.go

```diff
@@ -0,0 +1,69 @@
+//go:build js
+
+package js
+
+import (
+	"syscall/js"
+
+	"github.com/cosmos/gogoproto/proto"
+)
+
+// asyncHandler takes a handler and turns it into a JS function which returns a promise.
+// First a Promise executor function is created, inside of which a worker go routine is created and dispatched.
+// The executor returns immediately and the returned Promise is returned as the result of the function
+//
+// Inside the worker go routine, the Handler is invoked and according to the results
+// either invokes the reject if the error wasn't nil or resolves the promise with the result
+func asyncHandler[R proto.Message](handler func(js.Value, []js.Value) (R, error)) js.Func {
```

Comment:
move / refactor this guy. turn this into a function wrapper which takes a func(js.Value, []js.Value) (R, error) and returns a func(js.Value, []js.Value) (js.Value, error).
move to the proxy.